flink-runtime/src/main/java/org/apache/flink/runtime/metrics/TimerGauge.java

```diff
@@ -59,6 +59,27 @@ public synchronized void markEnd() {
         }
     }
 
+    /**
+     * Duplicate of {@link #markStart()} with ability passing the time from outside for possible
+     * optimization on calling {@link Clock#absoluteTimeMillis()}.
+     */
+    public synchronized void markStart(long absoluteTimeMillis) {
+        if (currentMeasurementStart == 0) {
+            currentMeasurementStart = absoluteTimeMillis;
+        }
+    }
```

Comment:
nit:
1. make this `private void markStartUnsafe(long absoluteTimeMillis);`
2. use this unsafe method in both `public synchronized void markEnd()` and `public synchronized void markStart(long absoluteTimeMillis)` to deduplicate the code?

ditto for `markEnd`?